test/e2e/suites/02-agreements.js

```diff
@@ -104,6 +111,40 @@ describe('Agreements suite', function AgreementSuite() {
       billingAgreement.id = result.id;
     });
 
+    // sorry for being verbose, will deal with it later
+    it('should bill agreement', async () => {
+      const { id } = billingAgreement;
+      // stub internal action call
+      const publishStub = sandbox.stub(payments.amqp, 'publish');
+      publishStub
+        .withArgs('payments.hook.publish', sinon.match.object, sinon.match.object)
+        .resolves();
+      publishStub.callThrough();
+
+      const result = await dispatch(billAgreement, { agreement: id, nextCycle: Date.now(), username: 'test@test.ru' });
+
+      assert.strictEqual(result, 'OK');
+
+      const hookPublishCalls = publishStub.getCalls().filter((call) => call.firstArg === 'payments.hook.publish');
+      assert.strictEqual(hookPublishCalls.length, 1);
+      sinon.assert.calledWithExactly(
```

Comment:
Think it will be easier `sinon.assert.calledWithExactly(publishStub....` 😀

Reply:
There is a number of other calls to publish, which the stub calls through, so I filter them first by the first argument. Then I assert on the filtered call only.